pipeline/metadata/beam_metadata.py

```diff
@@ -153,3 +170,37 @@ def merge_tagged_answers_with_rows(
       if tags.ip == untagged_answer.ip:
         merge_satellite_answers(untagged_answer, tags)
   return row
+
+
+def merge_blockpages_with_answers(
+    key: DomainDateIpKey,  # pylint: disable=unused-argument
+    value: Dict[str, Union[List[SatelliteAnswer],
+                           List[Tuple[str, SatelliteAnswerWithKeys]]]]
+) -> Iterator[Tuple[str, SatelliteAnswer]]:
+  """Add blockpage info to answers."""
+  answers: List[Tuple[str, SatelliteAnswer]] = value[
+      RECEIVED_IPS_PCOLLECTION_NAME]  # type: ignore
+  blockpages: List[BlockpageRow] = value[
+      BLOCKPAGE_PCOLLECTION_NAME]  # type: ignore
+
+  if not blockpages:
+    for (roundtrip_id, answer) in answers:
+      yield (roundtrip_id, answer)
+
+  https_blockpages = [
+      blockpage for blockpage in blockpages if blockpage.https is True
```

Comment:
I'd point out that you are flattening and merging http and https, just to separate them again here.
My recommendation is to keep them both in the same structure. There's no point in separating them. Or have them be separate inputs handled by separate functions.

The red flag here is that HTTP and HTTPS fetches here have different types, but are put in the same collection.

Reply:
They're the same type (in the same way that test rows and control rows are the same type, but have different `is_control` fields). But they could have something like a parent type which holds two of them.

But it did make more sense to separate them when we were writing them to their own table. Now with the inlining it's less useful.